src/com/xilinx/rapidwright/design/ModuleInst.java

```diff
@@ -358,6 +355,11 @@ public boolean place(Site newAnchorSite, boolean skipIncompatible, boolean allow
                 }
             }
 
+            if (existingSiteInst != null && !allowOverlap) {
+                unplace();
+                return false;
+            }
+
```

Comment:
It's not clear to me why this early exit is needed given the same action is performed on line 374--375 (which by observation would only trigger for `existingSiteInst != null && originalSites == null && !skipIncompatible`).

It no longer triggers on the latest version of the array generation code this was created for.

Reply:
If `skipIncompatible==true`, this needs to be checked earlier otherwise there is a failure.  It looks like in the array generation code this is true, but should probably be set to false.  It looks like in the latest version of the code, the stride has changed and does not trigger the issue.  I double checked and it still does for me in the initial version.